narwhal/consensus/src/tusk.rs

```diff
@@ -96,33 +93,27 @@ impl ConsensusProtocol for Tusk {
 
             // Starting from the oldest leader, flatten the sub-dag referenced by the leader.
             for x in utils::order_dag(self.gc_depth, leader, state) {
-                let digest = x.digest();
-
                 // Update and clean up internal state.
                 state.update(&x, self.gc_depth);
 
                 // Add the certificate to the sequence.
-                sequence.push(ConsensusOutput {
-                    certificate: x,
-                    consensus_index,
-                });
-
-                // Increase the global consensus index.
-                consensus_index += 1;
-
-                // Persist the update.
-                // TODO [issue #116]: Ensure this is not a performance bottleneck.
-                self.store.write_consensus_state(
-                    &state.last_committed,
-                    &consensus_index,
-                    &digest,
-                )?;
+                sequence.push(x);
             }
-
-            committed_sub_dags.push(CommittedSubDag {
+            let sub_dag = CommittedSubDag {
                 certificates: sequence,
                 leader: leader.clone(),
-            });
+                sub_dag_index: consensus_index,
+            };
+
+            // Increase the global consensus index.
+            consensus_index += 1;
```

Comment:
Btw, I am not sure what is the need for a global consensus index at this point?
For execution indexes should we just have something like `{commit_round, certificate_index_in_commit, batch_index_in_certificate, txn_index_in_batch}`?

Reply:
I am not sure it is needed at this point, but removing it may be easier on another PR because I think it will require changing the crash-recovery strategy

Reply:
I am also not sure we need the consensus store anymore

Reply:
Yes, I am planning to address all these in the next PR 👍